\documentclass{article}

\usepackage{pgfplotstable}
\usepackage{graphicx}


\DeclareGraphicsExtensions{.pdf,.png,.jpg}

\title{HiSeq X 30x vs 15x Comparison}
\author{Martin O. Pollard}


\begin{document}
  \pagenumbering{gobble}
  \maketitle
  \newpage
  \pagenumbering{arabic}
  \tableofcontents
  \newpage
  \section{Abstract}

  \section{Introduction}

  The cost of sequencing has decreased substanially in recent years, dropping to
  below \$1000 per sample if you exclude informatics and analysis costs. However
  the cost remains non-negliable and thus care must be taken in sample
  selection and choice of sequencing parameters such as intended coverage, in
  order to ensure the maximum amount of science is possible from what has been
  chosen.

  \section{Methods}
  100 samples from the general greek population were sequenced on the Illumina
  HiSeq X with an intended depth of 30x. The samples were aligned using BWA MEM
  0.7.8 to the 1000 Genomes hs37d5 reference; duplicates were then marked using
  biobambam and the files were stored in the sequencing IRODS.  These files were
  downloaded and downsampled to both 30x and 15x based on a genome size of 3.3 *
  10\^9 using samtools view. From this the downsampled BAMs where fed to GATK
  HaplotypeCaller 3.3 and per sample GVCF files were created for chromosome 6
  only. Lastly the GVCF files were combined and genotyped across all 100 samples
  using GATK GenotypeGVCFs to create two VCF files, one for the 15x data and one
  for the 30x data. As part of this process the VCF file's ID column is filled
  with IDs derived from DBSNP version 141.

  To provide us with additional information about the variants discovered the
  VCF we used bcftools annotate to add the 1000 Genomes Phase 3 European Allele
  Frequency information to each site.  Lastly in order to restrict variants
  sites to those that are highly likely to be real we filtered the sites in the
  VCF to those present in 1000 Genomes Phase 3 by using bcftools isec.

  \section{Results}
  After marking of reads with cycle INDEL bubbles several of the samples
  coverage fell below 30x. We ignored this and treated these samples as if they
  were 30x.
  \begin{table}[!h]
    \begin{center}
      \label{psc_table}
      \caption{Per sample paired comparisons}
      \pgfplotstabletypeset[
         col sep=tab,
         header=true,
         columns={field,30x mean,15x mean,mean diff,percent mean diff,p value},
         columns/field/.style={column name=Variant type, string type, column type=l},
         columns/30x mean/.style={fixed,precision=2, column type=r},
         columns/15x mean/.style={fixed,precision=2, column type=r},
         columns/mean diff/.style={fixed,precision=2, column type=r},
         columns/percent mean diff/.style={fixed,precision=3, column type=r},
         columns/p value/.style={sci,precision=3, column type=r},
       ]{data/analysis.tsv}
    \end{center}
  \end{table}

  \begin{figure}
    \caption{Chr 6 SNPs in also found 1KGP3 by allele frequency}
    \includegraphics[width=\textwidth]{data/in1kg_snp}
  \end{figure}
  \begin{figure}
    \caption{Chr 6 INDELs also found in 1KGP3 by allele frequency}
    \includegraphics[width=\textwidth]{data/in1kg_indel}
  \end{figure}

  \section{Acknowledgements}
  The GATK3 program was made available through the generosity of Medical and
  Population Genetics program at the Broad Institute, Inc.

  \newpage
  \section{Appendix A - Sample/Lane Characteristics}
  \begin{table}[!h]
    \begin{center}
      \label{sample_table}
      \caption{Sample characteristics (each lane contains one sample)}
      \pgfplotstabletypeset[
        col sep=tab,
        header=true,
        columns={lane,total reads,pass reads,fail reads},
        columns/lane/.style={column name=Lane, string type, column type=l},
      ]{data/lanestats.tsv}
    \end{center}
  \end{table}

\end{document}
